Crypt/GPG/Engine.php

```diff
@@ -1663,7 +1663,7 @@ private function _openSubprocess()
             $descriptorSpec,
             $this->_pipes,
             null,
-            $env,
+            array_filter($env, 'is_scalar'),
```

Comment:
I'd probably do this once, when `$env` is being set. And add a comment describing why it is needed.